quickstarts/microsoft.databasewatcher/watcher-new-datastore/azuredeploy.json

```diff
@@ -376,7 +376,7 @@
             "dependsOn": [
                 "[parameters('name')]"
             ],
-            "properties": "[json(parameters('targetProperties')[copyIndex('targetcopy')])]"
+            "properties": "[parameters('targetProperties')[copyIndex('targetcopy')]]"
```

Comment:
Instead of injecting both property name and value, you should only provide **_values_** with the parameters. Users consuming the template need to be able to see the properties being declared. In this case something like the following (assuming `targetProperties` changes to an object instead of an array with only one object in it):

```json
"properties": {
  "targetType": "[parameters('targetProperties').targetType]"
  "targetAuthenticationType": "[parameters('targetProperties').targetAuthenticationType]"
  "targetResourceId": "[resourceId('microsoft.sql/servers/databases', parameters('serverName'), parameters('databaseName'))]",
  ...
}
```

I wrote these expressions freehand, so they probably have errors, but hopefully this gets you on the right track

Reply:
The nuance here is that there might be multiple targets in the `targetProperties` array, each with their own distinct set of properties that would need to be parameterized. Is the pattern for that to introduce numbered parameters, i.e. `targetType1`, `targetType2`, etc?

While in the current example there is only one target, I would like to expand it later to show how to add multiple targets in one deployment.

Reply:
In that case, it is ok to have an array of objects that you iterate through, but it would be the same principal -- only parameterize the values, not the property names. You can do that in an ARM Template with the `copy` property and `copyIndex()` functions. TBH, that is not very fun ARM Template code to write, so I would strongly encourage using Bicep instead, but that is up to you :)